### PR TITLE
logging: Fix RTT backend drop mode and coverity issue

### LINKED ISSUE #15169: [Coverity CID :197534]Memory - corruptions in /subsys/logging/log_backend_rtt.c
Static code scan issues seen in File: /subsys/logging/log_backend_rtt.c
 Category: Memory - corruptions
 Function: line_out_drop_mode
 Component: Other
 CID: 197534
 Please fix or provide comments to square it off in coverity in the link: https://scan9.coverity.com/reports.htm#v32951/p12996

### PR DESCRIPTION
**Pull request by nordic-krch**
Drop mode in RTT backend was broken. It is not the default
one so most likely it was unnoticed for a long time. In order
to see issues user would need to pick this mode and have
enough logs to see data being dropped.
Fixed to report correctly dropped messages when logs are lost.

Additionally, added assert to ensure that memmove does not
do memory overwrite.

Fixes #15169.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

### CODE REVIEW

File: subsys/logging/log_backend_rtt.c [hunk 1]
```diff
@@ -108,18 +108,19 @@ static int char_out_drop_mode(u8_t data)
 
 static int line_out_drop_mode(void)
 {
-	*line_pos = '\r';
+	*(line_pos - 1) = '\r';
```
Comment (jarz-nordic) [inline subsys/logging/log_backend_rtt.c]: If there is only `\n` in the buffer this can cause writing outside of an array this pointer is pointing to.

Comment (nordic-krch) [reply subsys/logging/log_backend_rtt.c]: handled

[the remark this answers is not in the document]
Comment (nordic-krch) [reply subsys/logging/log_backend_rtt.c]: handled.

**Comment by nordic-krch** [comment]
@pizi-nordic @jarz-nordic i've resolved your comments. Can you take a look?

**Review by jakub-uC — APPROVED** [review]

**Review by pizi-nordic — APPROVED** [review]

**Pull request merged**